In [7]:
# Import all required modules
import pandas as pd
import numpy as np

In [3]:
#Reading the data
#NB: Make sure you the data in your current working directory

data = pd.read_csv('superstore data.csv',parse_dates = ['Order Date','Ship Date'],encoding = 'latin1')

In [4]:
data.head(3)

Row ID       Order ID Order Date  Ship Date       Ship Mode Customer ID  \
0   42433   AG-2011-2040 2011-01-01 2011-06-01  Standard Class    TB-11280   
1   22253  IN-2011-47883 2011-01-01 2011-08-01  Standard Class    JH-15985   
2   48883   HU-2011-1220 2011-01-01 2011-05-01    Second Class      AT-735   

     Customer Name   Segment         City            State  ...  \
0  Toby Braunhardt  Consumer  Constantine      Constantine  ...   
1      Joseph Holt  Consumer  Wagga Wagga  New South Wales  ...   
2    Annie Thurman  Consumer     Budapest         Budapest  ...   

         Product ID         Category Sub-Category              Product Name  \
0  OFF-TEN-10000025  Office Supplies      Storage       Tenex Lockers, Blue   
1   OFF-SU-10000618  Office Supplies     Supplies  Acme Trimmer, High Speed   
2  OFF-TEN-10001585  Office Supplies      Storage   Tenex Box, Single Width   

     Sales Quantity Discount   Profit  Shipping Cost  Order Priority  
0  408.300        2      0.0  106.140          35.46          Medium  
1  120.366        3      0.1   36.036           9.72          Medium  
2   66.120        4      0.0   29.640           8.17            High  

[3 rows x 24 columns]

In [5]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51290 entries, 0 to 51289
Data columns (total 24 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   Row ID          51290 non-null  int64         
 1   Order ID        51290 non-null  object        
 2   Order Date      51290 non-null  datetime64[ns]
 3   Ship Date       51290 non-null  datetime64[ns]
 4   Ship Mode       51290 non-null  object        
 5   Customer ID     51290 non-null  object        
 6   Customer Name   51290 non-null  object        
 7   Segment         51290 non-null  object        
 8   City            51290 non-null  object        
 9   State           51290 non-null  object        
 10  Country         51290 non-null  object        
 11  Postal Code     9994 non-null   float64       
 12  Market          51290 non-null  object        
 13  Region          51290 non-null  object        
 14  Product ID      51290 non-null  object        
 15  Ca

In [8]:
len(np.sort(data['Country'].unique()))

147

## Setting Keys. 
Country is set as an index

In [9]:
data_copy_countries = data.copy()
data_copy_markets = data.copy()
data_copy_category = data.copy()
data_copy_countries.set_index('Country',inplace = True)           #set country as an index
data_copy_markets.set_index('Market',inplace = True)           #set market as an index

data_copy_category.set_index('Category',inplace = True)           #set product category as an index 

In [10]:
data_copy_markets.head()

Row ID         Order ID Order Date  Ship Date       Ship Mode  \
Market                                                                  
Africa   42433     AG-2011-2040 2011-01-01 2011-06-01  Standard Class   
APAC     22253    IN-2011-47883 2011-01-01 2011-08-01  Standard Class   
EMEA     48883     HU-2011-1220 2011-01-01 2011-05-01    Second Class   
EU       11731  IT-2011-3647632 2011-01-01 2011-05-01    Second Class   
APAC     22255    IN-2011-47883 2011-01-01 2011-08-01  Standard Class   

       Customer ID    Customer Name      Segment         City  \
Market                                                          
Africa    TB-11280  Toby Braunhardt     Consumer  Constantine   
APAC      JH-15985      Joseph Holt     Consumer  Wagga Wagga   
EMEA        AT-735    Annie Thurman     Consumer     Budapest   
EU        EM-14140     Eugene Moren  Home Office    Stockholm   
APAC      JH-15985      Joseph Holt     Consumer  Wagga Wagga   

                  State  ...        Product ID         Category Sub-Category  \
Market                   ...                                                   
Africa      Constantine  ...  OFF-TEN-10000025  Office Supplies      Storage   
APAC    New South Wales  ...   OFF-SU-10000618  Office Supplies     Supplies   
EMEA           Budapest  ...  OFF-TEN-10001585  Office Supplies      Storage   
EU            Stockholm  ...   OFF-PA-10001492  Office Supplies        Paper   
APAC    New South Wales  ...   FUR-FU-10003447        Furniture  Furnishings   

                       Product Name    Sales Quantity Discount   Profit  \
Market                                                                    
Africa          Tenex Lockers, Blue  408.300        2      0.0  106.140   
APAC       Acme Trimmer, High Speed  120.366        3      0.1   36.036   
EMEA        Tenex Box, Single Width   66.120        4      0.0   29.640   
EU      Enermax Note Cards, Premium   44.865        3      0.5  -26.055   
APAC     Eldon Light Bulb, Duo Pack  113.670        5      0.1   37.770   

        Shipping Cost  Order Priority  
Market                                 
Africa          35.46          Medium  
APAC             9.72          Medium  
EMEA             8.17            High  
EU               4.82            High  
APAC             4.70          Medium  

[5 rows x 23 columns]

In [11]:
for col in data.columns:
    print(col)

Row ID
Order ID
Order Date
Ship Date
Ship Mode
Customer ID
Customer Name
Segment
City
State
Country
Postal Code
Market
Region
Product ID
Category
Sub-Category
Product Name
Sales
Quantity
Discount
Profit
Shipping Cost
Order Priority


### Hints on Data preparation and exploration 

1. Load the data.
2. Inspect the data
3. Check for missing values
4. Check the data of each column
5. Check the different categories in the categorical variables
6. Take a view of the data
7. Take the descritive statistics for the numerical variables
9. Check the ditribution of the variables
10. Feature Engineering
#Create columns for:

#Unit price (sale/quantity)

#Year

#Month

#interval between the order date and the shipping date in days.


## Hints on the analysis 

### Check: The following  possible questions may of interest.

1. **What is the sales and profit per market?**

2. **What is the most and least profitable market?**

3. How are you going to increase the sales for the products that least bought?

4. Which month does the company records more sales?

5. What is the correlation between discount and sale?

6. Does the shipping cost affect the sales?

7. What is the dominant product category in each market?

8. City with most orders for a given product category?

9. which products are making loss?

10. **Which segment provides more profit?**

11. what is the correlation between quantity and profit?

12. what it the correlation between shipping cost and profit?

13. Which order priority yields more sales?

14. What is the average shipping cost to each region?

15. What product categories should be targetted to the  different markets?

16. Does the interval between the order date and the shipping date affect sale and profit


# DATA PREPARATION

- inspecting data

In [12]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51290 entries, 0 to 51289
Data columns (total 24 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   Row ID          51290 non-null  int64         
 1   Order ID        51290 non-null  object        
 2   Order Date      51290 non-null  datetime64[ns]
 3   Ship Date       51290 non-null  datetime64[ns]
 4   Ship Mode       51290 non-null  object        
 5   Customer ID     51290 non-null  object        
 6   Customer Name   51290 non-null  object        
 7   Segment         51290 non-null  object        
 8   City            51290 non-null  object        
 9   State           51290 non-null  object        
 10  Country         51290 non-null  object        
 11  Postal Code     9994 non-null   float64       
 12  Market          51290 non-null  object        
 13  Region          51290 non-null  object        
 14  Product ID      51290 non-null  object        
 15  Ca

# sales and profit per area
- What is the sales and profit per market?
- What is the most and least profitable market?
- Which segment provides more profit?

### 1. What is the sales and profit per market?

In [13]:
# SALES AND PROFITS IN COUNTRIES

country_sales_sum = data_copy_countries['Sales'].groupby('Country').sum() #count sales in
                                                                          #every country
country_sales_mean = data_copy_countries['Sales'].groupby('Country').mean() #mean count 
                                                                            #sales in every country
    
country_profit_sum = data_copy_countries['Profit'].groupby('Country').sum() #count profit in
                                                                            #every country
country_profit_mean = data_copy_countries['Profit'].groupby('Country').mean() #mean count 
                                                                            #profit in every country

In [14]:
# SALES AND PROFITS IN MARKETS


market_sales_sum = data_copy_markets['Sales'].groupby('Market').sum() #count sales in
                                                                          #every country
market_sales_mean = data_copy_markets['Sales'].groupby('Market').mean() #mean count 
                                                                            #sales in every country
    
market_profit_sum = data_copy_markets['Profit'].groupby('Market').sum() #count profit in
                                                                          #every country
market_profit_mean = data_copy_markets['Profit'].groupby('Market').mean() #mean count 
                                                                            #profit in every country


In [15]:
#print('Countries by Average sales and profit:')

avg_sals = country_sales_mean.sort_values(ascending=False)  # Sort countries according to mean sales                                                            
total_sals = country_sales_sum.sort_values(ascending=False) # Sort countries according to total sales 
avg_prof = country_profit_mean.sort_values(ascending=False)  # Sort countries according to mean sales
total_prof = country_profit_sum.sort_values(ascending=False) # Sort countries according to total sales

In [16]:
#print('Markets by Average sales and profit:')

avg_sals = market_sales_mean.sort_values(ascending=False)  # Sort countries according to mean sales                                                          
total_sals = market_sales_sum.sort_values(ascending=False) # Sort countries according to total sales 
avg_prof = market_profit_mean.sort_values(ascending=False)  # Sort countries according to mean sales
total_prof = market_profit_sum.sort_values(ascending=False) # Sort countries according to total sales


### * Analysis on Market based on profit and sales

In [22]:

# Market information (Average and Total). Note that it is sorted according to profit

sales_profit_mkt = data_copy_markets[['Sales', 'Profit']]

def Market_info_avg(df):
    
    market_means = df.groupby('Market').mean() #mean count sales in every market
        
    market_means_sort = market_means.sort_values('Profit',ascending=False)  # Sort market according to mean sales


    return print(market_means_sort)

  
    
def Market_info_sum(df):
    
        
    market_sums = df.groupby('Market').sum() #count profit in
                                                                          #every market
    
    market_sums_sort = market_sums.sort_values('Profit',ascending=False)  # Sort market according to mean sales
    

    return print(market_sums_sort)   



In [23]:
Market_info_avg(sales_profit_mkt)

             Sales     Profit
Market                       
Canada  174.292109  46.399453
APAC    325.917481  39.629163
EU      293.808906  37.282974
US      229.858001  28.656896
LATAM   210.278334  21.531328
Africa  170.868370  19.374674
EMEA    160.302508   8.728966


In [24]:
Market_info_sum(sales_profit_mkt)

               Sales        Profit
Market                            
APAC    3.585744e+06  436000.04900
EU      2.938089e+06  372829.74150
US      2.297201e+06  286397.02170
LATAM   2.164605e+06  221643.48708
Africa  7.837732e+05   88871.63100
EMEA    8.061613e+05   43897.97100
Canada  6.692817e+04   17817.39000


### * Analysis on Country based on profit and sales

In [25]:
# Country information (Average and Total). Note that it is sorted according to profit

sales_profit_ctry = data_copy_countries[['Sales', 'Profit']]

def Country_info_avg():
    
    country_means = sales_profit_ctry.groupby('Country').mean() #mean count 
                                                                           #sales in every country
        
    country_means_sort = country_means.sort_values('Profit',ascending=False)  # Sort country according to mean sales


    return print(country_means_sort)

  
    
def Country_info_sum():
    
        
    country_sums = sales_profit_ctry.groupby('Country').sum() #count profit in every country
    
    country_sums_sort = country_sums.sort_values('Profit',ascending=False)  # Sort country according to mean sales
    

    return print(country_sums_sort)   


In [26]:
Country_info_avg()


                    Sales      Profit
Country                              
Montenegro    1001.092500  321.690000
Lesotho       1118.665000  233.035000
Slovenia       489.980000  196.680000
Mauritania     402.093000  167.595000
Taiwan         546.259286  129.900000
...                   ...         ...
Netherlands    178.195276  -94.413966
Yemen           82.190400 -123.548600
Portugal       215.787429 -124.329429
Turkmenistan    81.082440 -132.108360
Lithuania      134.486640 -179.322360

[147 rows x 2 columns]


In [27]:
Country_info_sum()

                       Sales       Profit
Country                                  
United States   2.297201e+06  286397.0217
China           7.005620e+05  150683.0850
India           5.896501e+05  129071.8350
United Kingdom  5.285763e+05  111900.1500
France          8.589311e+05  109029.0030
...                      ...          ...
Pakistan        5.887261e+04  -22446.6480
Honduras        9.012565e+04  -29482.3714
Netherlands     7.751494e+04  -41070.0750
Nigeria         5.435035e+04  -80750.7180
Turkey          1.085079e+05  -98447.2320

[147 rows x 2 columns]


### 2.  **What is the most and least profitable market?**

In [28]:
sals_profit_mkt = data_copy_markets[['Sales', 'Profit', 'Category']]

market_mean = sales_profit_mkt.groupby('Market').mean() #mean count 
market_means_sort = market_mean.sort_values('Profit',ascending=False)  # Sort market according to mean sales

In [29]:
sals_profit_mkt

Sales    Profit         Category
Market                                    
Africa  408.300  106.1400  Office Supplies
APAC    120.366   36.0360  Office Supplies
EMEA     66.120   29.6400  Office Supplies
EU       44.865  -26.0550  Office Supplies
APAC    113.670   37.7700        Furniture
...         ...       ...              ...
US       13.904    4.5188  Office Supplies
Africa    3.990    0.4200  Office Supplies
LATAM    26.400   12.3600  Office Supplies
LATAM     7.120    0.5600  Office Supplies
US        3.024   -0.6048  Office Supplies

[51290 rows x 3 columns]

In [30]:
sales_profit_cat = data_copy_category[['Sales', 'Profit']]


In [31]:
cat_mean = sales_profit_cat.groupby('Category').mean() #mean count 
market_means_sort = cat_mean.sort_values('Profit',ascending=False)  # Sort market according to mean sales

In [32]:
market_means_sort

Sales     Profit
Category                              
Technology       467.858939  65.454958
Furniture        416.248905  28.878567
Office Supplies  121.097120  16.578961

In [33]:
#Analyzing sales and profit (On Average) in Markets based on category

Technology_Avg = data_copy_markets[data_copy_markets['Category']=='Technology'][['Sales', 'Profit']
                                                                    ].groupby('Market').mean(
                                                                    ).sort_values('Profit',
                                                                                 ascending=False)
                                                                    # Sorting results based on profit

Furniture_Avg = data_copy_markets[data_copy_markets['Category']=='Furniture'][['Sales', 'Profit']
                                                                    ].groupby('Market').mean(
                                                                    ).sort_values('Profit',
                                                                                 ascending=False)
                                                                    # Sorting results based on profit
Office_Supplies_Avg = data_copy_markets[data_copy_markets['Category']=='Office Supplies'][['Sales', 'Profit'
                                                                    ]].groupby('Market').mean(
                                                                     ).sort_values('Profit',
                                                                                 ascending=False)
                                                                    # Sorting results based on profit


cat_mkt_sal_prof_avg = pd.concat([Technology_Avg,Furniture_Avg,Office_Supplies_Avg],keys=
                                      ['Technology','Furniture','Office_Supplies'])

#'''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
#Analyzing sales and profit (In Total) in Markets based on category

Technology_sum = data_copy_markets[data_copy_markets['Category']=='Technology'][['Sales', 'Profit']
                                                                    ].groupby('Market').sum(
                                                                    ).sort_values('Profit',
                                                                                 ascending=False)
                                                                    # Sorting results based on profit

Furniture_sum = data_copy_markets[data_copy_markets['Category']=='Furniture'][['Sales', 'Profit']
                                                                    ].groupby('Market').sum(
                                                                    ).sort_values('Profit',
                                                                                 ascending=False)
                                                                    # Sorting results based on profit
Office_Supplies_sum = data_copy_markets[data_copy_markets['Category']=='Office Supplies'][['Sales', 'Profit'
                                                                    ]].groupby('Market').sum(
                                                                     ).sort_values('Profit',
                                                                                 ascending=False)
                                                                    # Sorting results based on profit


cat_mkt_sal_prof_sum = pd.concat([Technology_sum,Furniture_sum,Office_Supplies_sum],keys=
                                      ['Technology','Furniture','Office_Supplies'])



In [34]:
cat_mkt_sal_prof_sum

Sales        Profit
                Market                            
Technology      APAC    1.357490e+06  204593.85600
                US      8.361540e+05  145454.94810
                EU      1.112679e+06  144272.65800
                LATAM   7.887138e+05  100586.71508
                Africa  3.223670e+05   44129.49300
                EMEA    3.008546e+05   17494.44300
                Canada  2.629881e+04    7246.62000
Furniture       APAC    1.343934e+06  125896.15050
                EU      7.791030e+05   68850.13950
                LATAM   8.119706e+05   41597.70000
                US      7.419998e+05   18451.27280
                Africa  1.946506e+05   16262.08500
                EMEA    2.286207e+05   11534.13600
                Canada  1.059528e+04    2613.24000
Office_Supplies EU      1.046307e+06  159706.94400
                US      7.190470e+05  122490.80080
                APAC    8.843198e+05  105510.04250
                LATAM   5.639207e+05   79459.07200
                Africa  2.667555e+05   28480.05300
                EMEA    2.766860e+05   14869.39200
                Canada  3.003408e+04    7957.53000

## QUESTION: What is the most and least profitable market?
Most profitable market

- Most profitable, On average : Canada  (**46.399453**)
- Most profitable, In Total : APAC    (**436000.04900**)


Least profitable market

- Least profitable, On average : EMEA     (**8.728966**) @ a sale of **160.302508 **
- Least profitable, In Total : Canada     (**17817.39000**)  @ a sale of  66,928.17   


**Group Question: What about, in terms of product category?**

- **Technology**: 

 - Most profitable, On average : Canada 	(**111.486462**) @ a sale of **404.597077 units**
 - Most profitable, In Total : APAC   (**204593.85600**) @ a sale of  ***1,357,490** units

 - Least profitable, On average : EMEA 	(**18.185492**) @ a sale of  ***312.738652** units
 - Least profitable, In Total :  Canada 	(**7246.62000**) @ a sale of  ***2,629.881** units
 
- **Furniture**:

 - Most profitable, On average : Canada 	(**62.220000**) @ a sale of **252.268571 units**
 - Most profitable, In Total : APAC 	(**125896.15050**) @ a sale of **13,43,934 units**

 - Least profitable, On average : US 	(**8.699327**) @ a sale of  ***349.834887** units
 - Least profitable, In Total :  Canada 	(**2613.24000**) @ a sale of  ***10,595.28e+04** units
 
- **Office_Supplies**:

 - Most profitable, On average : Canada 	(**28.727545**) @ a sale of **108.426282 units**
 - Most profitable, In Total : EU 	(**159706.94400**) @ a sale of **1,046,307 units**
 Canada 	3.003408e+04 	7957.53000 
 - Least profitable, On average : EMEA 	(**4.509976**) @ a sale of  ***83.920531** units
 - Least profitable, In Total :  Canada 	(**7957.53000**) @ a sale of  ***30034.08** units

## 3. How are you going to increase the sales for the products that least bought?

In [35]:
data.head()

Row ID         Order ID Order Date  Ship Date       Ship Mode Customer ID  \
0   42433     AG-2011-2040 2011-01-01 2011-06-01  Standard Class    TB-11280   
1   22253    IN-2011-47883 2011-01-01 2011-08-01  Standard Class    JH-15985   
2   48883     HU-2011-1220 2011-01-01 2011-05-01    Second Class      AT-735   
3   11731  IT-2011-3647632 2011-01-01 2011-05-01    Second Class    EM-14140   
4   22255    IN-2011-47883 2011-01-01 2011-08-01  Standard Class    JH-15985   

     Customer Name      Segment         City            State  ...  \
0  Toby Braunhardt     Consumer  Constantine      Constantine  ...   
1      Joseph Holt     Consumer  Wagga Wagga  New South Wales  ...   
2    Annie Thurman     Consumer     Budapest         Budapest  ...   
3     Eugene Moren  Home Office    Stockholm        Stockholm  ...   
4      Joseph Holt     Consumer  Wagga Wagga  New South Wales  ...   

         Product ID         Category Sub-Category  \
0  OFF-TEN-10000025  Office Supplies      Storage   
1   OFF-SU-10000618  Office Supplies     Supplies   
2  OFF-TEN-10001585  Office Supplies      Storage   
3   OFF-PA-10001492  Office Supplies        Paper   
4   FUR-FU-10003447        Furniture  Furnishings   

                  Product Name    Sales Quantity Discount   Profit  \
0          Tenex Lockers, Blue  408.300        2      0.0  106.140   
1     Acme Trimmer, High Speed  120.366        3      0.1   36.036   
2      Tenex Box, Single Width   66.120        4      0.0   29.640   
3  Enermax Note Cards, Premium   44.865        3      0.5  -26.055   
4   Eldon Light Bulb, Duo Pack  113.670        5      0.1   37.770   

   Shipping Cost  Order Priority  
0          35.46          Medium  
1           9.72          Medium  
2           8.17            High  
3           4.82            High  
4           4.70          Medium  

[5 rows x 24 columns]

In [36]:
groups_1 = data[['Category', 'Shipping Cost', 'Ship Mode', 'Market', 'Discount', 'Sales']].sort_values('Sales', ascending=True).groupby('Market')

In [ ]:
for mkt,mkt_data in groups_1:
    print()
    print(mkt)
    print()
    print(mkt_data)

In [ ]:
for col in data.columns:
    print(col)

In [ ]:
#Identifying order priority
order_sals=data[['Order Priority', 'Sales']].sort_values('Sales', ascending = True).groupby('Order Priority')

for col, xx in order_sals:
    print()
    print(col)
    print()
    print(xx)

In [ ]:
# Identify the products in each category. Discount is added
# How can I extract columns individually in this?
# Establish relationships between discount and sales column to determine how they're interacting
# This will be possible after pulling out the columns in each category.
order_product_prof=data[['Order Priority','Category', 'Discount', 
                         'Sales']].sort_values('Sales', ascending = True).groupby('Order Priority')


for col, xx in order_product_prof:
    print()
    print(col)
    print()
    print(xx)

In [ ]:
# Considering Regions and their sales
reg = data[['Region', 'Product Name', 'Sales']].groupby('Region').sum().sort_values('Sales', ascending=False)
reg

In [ ]:
# Taking Various regions against their sales

sals_in_reg = data[['Region', 'Sales']]       
sals_in_reg[sals_in_reg['Region']=='Central' ].sort_values('Sales') # Central

sals_in_reg[sals_in_reg['Region']=='South' ].sort_values('Sales') # South

sals_in_reg[sals_in_reg['Region']=='North' ].sort_values('Sales') # North

sals_in_reg[sals_in_reg['Region']=='Oceania' ].sort_values('Sales') # Oceania

sals_in_reg[sals_in_reg['Region']=='Southeast Asia' ].sort_values('Sales') # Southeast Asia

sals_in_reg[sals_in_reg['Region']=='North Asia' ].sort_values('Sales') # North Asia

sals_in_reg[sals_in_reg['Region']=='EMEA' ].sort_values('Sales') # EMEA

sals_in_reg[sals_in_reg['Region']=='Africa' ].sort_values('Sales') # Africa

sals_in_reg[sals_in_reg['Region']=='Central Asia' ].sort_values('Sales') # Central Asia

sals_in_reg[sals_in_reg['Region']=='West' ].sort_values('Sales') # West

sals_in_reg[sals_in_reg['Region']=='East' ].sort_values('Sales') # East

sals_in_reg[sals_in_reg['Region']=='Caribbean' ].sort_values('Sales') # Caribbean

sals_in_reg[sals_in_reg['Region']=='Canada' ].sort_values('Sales') # Canada

## 3. Which month does the company records more sales?

In [ ]:
month_sals = data[['Ship Date', 'Market','Category', 'Sales']].sort_values('Sales',ascending=False)  # Sort market according to mean sales

In [ ]:
month_sals

**March** had the highest sales of **22638.480 Units**

**Group Questions**
- In which Market:  **US**
- What product category had the highest sales in March: **Technology**

## 5. What is the correlation between discount and sale?

In [ ]:
#sal['title_len'] = sal['JobTitle'].apply(len)

data_corr = data[['Discount', 'Sales', 'Market', 'Category']]
disc_sals_corr = data_corr[['Discount', 'Sales']].corr()
disc_sals_corr

In [ ]:
plt.scatter(data_corr[['Discount']],data_corr[['Sales']])
plt.show

There is a week negative correlation of **-0.087** between Discount and sales


**For Markets:**

In [ ]:
data_mkt = data_copy_markets[['Discount', 'Sales']] 
data_mkt_index = data_mkt.index.sort_values().unique()
data_mkt_index

**For Category:**

In [ ]:
data_cat = data_copy_category[['Discount', 'Sales']] 
data_cat_index = data_cat.index.sort_values().unique()
data_cat_index

**1. Determining the correlation between discount and sales in each market**


In [ ]:
# 1. Determining the correlation between discount and sales in each market

data_APAC = data_mkt.loc[['APAC']][['Discount', 'Sales']].corr()
data_Africa = data_mkt.loc[['Africa']][['Discount', 'Sales']].corr()
data_Canada = data_mkt.loc[['Canada']][['Discount', 'Sales']].corr()
data_EMEA = data_mkt.loc[['EMEA']][['Discount', 'Sales']].corr()
data_EU = data_mkt.loc[['EU']][['Discount', 'Sales']].corr()
data_LATAM = data_mkt.loc[['LATAM']][['Discount', 'Sales']].corr()
data_US = data_mkt.loc[['US']][['Discount', 'Sales']].corr()


#Inspecting Discount and Sales in every Market

prof_sal_markt_1 = pd.concat([data_mkt.loc[['APAC']],data_mkt.loc[['Africa']],
                            data_mkt.loc[['Canada']],data_mkt.loc[['EMEA']],
                            data_mkt.loc[['EU']],data_mkt.loc[['LATAM']],
                            data_mkt.loc[['US']]],keys=['APAC','Africa','Canada','EMEA','EU',
               'LATAM','US'])

prof_sal_markt_inspect = pd.concat([data_mkt.loc[['APAC']].max(),data_mkt.loc[['Africa']].max(),
                            data_mkt.loc[['Canada']].max(),data_mkt.loc[['EMEA']].max(),
                            data_mkt.loc[['EU']].max(),data_mkt.loc[['LATAM']].max(),
                            data_mkt.loc[['US']].max()],keys=['APAC','Africa','Canada','EMEA','EU',
               'LATAM','US'])

# Generating the correlation 
corr_market = pd.concat([data_APAC,data_Africa,data_Canada,data_EMEA,data_EU,
               data_LATAM,data_US],keys=['APAC','Africa','Canada','EMEA','EU',
               'LATAM','US'])


In [ ]:
prof_sal_markt_inspect # maimum Discout and maximum Sales in each region

In [ ]:
corr_market

Negative correlation between Discount and Sales existed in all the markets.
However, no correlation between discount and sales was established since the company issued no discount in Canada.

**2. Determining the correlation between discount and sales in each market**


In [ ]:
data_cat.index.unique()

In [ ]:
data_off_sup = data_cat.loc[['Office Supplies']][['Discount', 'Sales']].corr()
data_furn = data_cat.loc[['Furniture']][['Discount', 'Sales']].corr()
data_techno = data_cat.loc[['Technology']][['Discount', 'Sales']].corr()



#Inspecting Discount and Sales in every Market

prof_sal_cat_1 = pd.concat([data_cat.loc[['Office Supplies']],data_cat.loc[['Furniture']],
                            data_cat.loc[['Technology']]],
                             keys=['Office Supplies','Furniture','Technology'])

prof_sal_cat_inspect = pd.concat([data_cat.loc[['Office Supplies']].max(),data_cat.loc[['Furniture']].max(),
                            data_cat.loc[['Technology']][['Discount']].max()],keys=['Office Supplies','Furniture','Technology'])

#Generating the correlation 
corr_cat = pd.concat([data_off_sup,data_furn,data_techno],keys=['Office Supplies','Furniture','Technology'])


In [ ]:
prof_sal_cat_inspect      # maximum Discount and maximum sales of each product category

In [ ]:
corr_cat

Negative correlation between Discount and Sales existed in all product categories.


In [ ]:
data['Ship Mode'].unique()

In [ ]:
data.head()

In [21]:
data['Sales'].corr(data['Shipping Cost'])

0.7680728376889517

## QUESTION 6: Does the shipping cost affect the sales?

Correlation of 0.77 was found between Sales and Shipping cost so Shipping cost and sales affect each other

In [22]:
group = data.groupby(["Market","Category"]).agg({'Profit': 'sum'})
group[['Profit']]

Profit
Market Category                     
APAC   Furniture        125896.15050
       Office Supplies  105510.04250
       Technology       204593.85600
Africa Furniture         16262.08500
       Office Supplies   28480.05300
       Technology        44129.49300
Canada Furniture          2613.24000
       Office Supplies    7957.53000
       Technology         7246.62000
EMEA   Furniture         11534.13600
       Office Supplies   14869.39200
       Technology        17494.44300
EU     Furniture         68850.13950
       Office Supplies  159706.94400
       Technology       144272.65800
LATAM  Furniture         41597.70000
       Office Supplies   79459.07200
       Technology       100586.71508
US     Furniture         18451.27280
       Office Supplies  122490.80080
       Technology       145454.94810

In [23]:
group = data.groupby(["Market","Category"]).agg({'Sales': 'sum'})
group[['Sales']]

Sales
Market Category                     
APAC   Furniture        1.343934e+06
       Office Supplies  8.843198e+05
       Technology       1.357490e+06
Africa Furniture        1.946506e+05
       Office Supplies  2.667555e+05
       Technology       3.223670e+05
Canada Furniture        1.059528e+04
       Office Supplies  3.003408e+04
       Technology       2.629881e+04
EMEA   Furniture        2.286207e+05
       Office Supplies  2.766860e+05
       Technology       3.008546e+05
EU     Furniture        7.791030e+05
       Office Supplies  1.046307e+06
       Technology       1.112679e+06
LATAM  Furniture        8.119706e+05
       Office Supplies  5.639207e+05
       Technology       7.887138e+05
US     Furniture        7.419998e+05
       Office Supplies  7.190470e+05
       Technology       8.361540e+05

## QUESTION 7: What is the dominant product category in each market?

**Group Question: What about, in terms of product category?**

- **APAC**: 

 - Most Sales: Technology (**1.357490e+06**)
 - Most profitable: Technology   (**204593.85600**)

 - Least Sales: Office Supplies (**8.843198e+05**)
 - Least profitable: Office Supplies   (**105510.04250**)
    
- **Africa**: 

 - Most Sales: Technology (**3.223670e+05**)
 - Most profitable: Technology   (**44129.49300**)

 - Least Sales: Furniture (**1.946506e+05**)
 - Least profitable: Furniture (**16262.08500**)
    
- **Canada**: 

 - Most Sales: Office Supplies (**3.003408e+04**)
 - Most profitable: Office Supplies (**7957.53000**)

 - Least Sales: Furniture (**1.059528e+04**)
 - Least profitable: Furniture (**2613.24000**)
    
- **EMEA**: 

 - Most Sales: Technology (**17494.44300**)
 - Most profitable: Technology (**17494.44300**)

 - Least Sales: Furniture (**3.008546e+05**)
 - Least profitable: Furniture (**11534.13600**)
    
- **EU**: 

 - Most Sales: Technology (**1.112679e+06**)
 - Most profitable: Office Supply (**159706.94400**)

 - Least Sales: Furniture (**7.791030e+05**)
 - Least profitable: Furniture (**68850.13950**)    
    
- **LATAM**: 

 - Most Sales: Furniture (**8.119706e+05**)
 - Most profitable: Technology (**100586.71508**)

 - Least Sales: Office Supply (**5.639207e+05**)
 - Least profitable: Furniture (**41597.70000**)
    
- **US**: 

 - Most Sales: Technology (**8.361540e+05**)
 - Most profitable: Technology (**145454.94810**)

 - Least Sales: Office Supply (**7.190470e+05**)
 - Least profitable: Furniture (**18451.27280**)
    

In [17]:
group = data.groupby(["Category","City"]).count()
group.columns
total_price = group["Order ID"].groupby(level=0, group_keys=False)
total_price.nlargest(5)

Category         City         
Furniture        New York City    192
                 Los Angeles      154
                 Philadelphia     111
                 Santo Domingo    105
                 Manila           104
Office Supplies  New York City    552
                 Los Angeles      443
                 San Francisco    322
                 Philadelphia     312
                 Santo Domingo    250
Technology       New York City    171
                 Los Angeles      150
                 Philadelphia     114
                 Manila           101
                 Santo Domingo     88
Name: Order ID, dtype: int64

## QUESTION 8: City with most orders for a given product category

New York City seems to top with most orders in each category

In [18]:
data[data['Profit'] < 0][["Product Name", "Profit"]].sort_values(by = "Profit").head()

Product Name     Profit
43453  Cubify CubeX 3D Printer Double Head Print -6599.9780
43136             Motorola Smart Phone, Cordless -4088.3760
15886  Cubify CubeX 3D Printer Triple Head Print -3839.9904
42767  GBC DocuBind P400 Electric Binding System -3701.8928
28813  Lexmark MX611dhe Monochrome Laser Printer -3399.9800

## QUESTION 9: which products are making loss?

Cubify CubeX 3D Printer Double Head Print	-6599.9780

In [19]:
data.groupby(['Product Name']).agg({'Profit': 'min'}).sort_values(by = "Profit", ascending = True).head()

Profit
Product Name                                        
Cubify CubeX 3D Printer Double Head Print -6599.9780
Motorola Smart Phone, Cordless            -4088.3760
Cubify CubeX 3D Printer Triple Head Print -3839.9904
GBC DocuBind P400 Electric Binding System -3701.8928
Lexmark MX611dhe Monochrome Laser Printer -3399.9800

## QUESTION 9: which products are making loss? --- Just another way of answering question

In [20]:
data.groupby(['Segment']).agg({'Profit': 'sum', 'Sales': 'mean'})

Profit       Sales
Segment                              
Consumer     749239.78206  245.416299
Corporate    441208.32866  247.890176
Home Office  277009.18056  247.228403

## QUESTION 10: Which segment provides more profit?

Consumer	749239.78206

In [24]:
data[['Quantity','Profit']].corr()

Quantity    Profit
Quantity  1.000000  0.104365
Profit    0.104365  1.000000

## QUESTION 11 : what is the correlation between quantity and profit?

Correlation of 0.1 means there is no correlation between quantity and profit

In [33]:
data[['Shipping Cost','Profit']].corr()

Shipping Cost    Profit
Shipping Cost       1.000000  0.354441
Profit              0.354441  1.000000

## QUESTION 12 : what it the correlation between shipping cost and profit?

Correlation of 0.3 means there is no correlation between Shippinng cost and profit

In [26]:
order_sales = data.groupby(['Order Priority']).agg({'Sales':'max'}).sort_values(by = 'Sales', ascending = False)
order_sales

Sales
Order Priority           
Medium          22638.480
High            11199.968
Critical         8749.950
Low              7958.580

## QUESTION 13: Which order priority yields more sales

Medium Order Priority yields more Sales and Low Priority yields the least

In [28]:
ship_cost_region = data.groupby(['Region']).agg({'Shipping Cost':'mean'}).sort_values(by = 'Shipping Cost', ascending = False)
ship_cost_region

Shipping Cost
Region                       
North Asia          40.646112
Central Asia        38.003003
Oceania             34.593780
Southeast Asia      29.937948
North               27.476299
Central             26.710494
East                25.542001
South               25.353509
West                23.887687
Caribbean           21.059260
Canada              19.285495
Africa              19.215058
EMEA                17.573221

## QUESTION 14: What is the average shipping cost to each region?

North Asia has the highest average shipping cost 40.646112 and EMEA has the least 17.573221

In [32]:
order_market = pd.DataFrame(data.groupby('Market').max()).sort_values('Profit',ascending = False)
order_market["Profit## QUESTION: Which order priority yields more sales

Medium Order Priority yields more Sales and Low Priority yields the least"]

Market
US        8399.976
EU        3979.080
APAC      2939.310
Africa    2597.280
EMEA      1622.880
LATAM     1313.280
Canada    1159.020
Name: Profit, dtype: float64

## QUESTION 15: What product categories should be targetted to the different markets?

North Asia has the highest average shipping cost 40.646112 and EMEA has the least 17.573221